### PR TITLE
Add pricing selection in DeployRequest and MigrationRequest

### LINKED ISSUE #1590: Add pricing selection in DeployRequest and MigrationRequest
- [ ] add a new mandatory `BillingMode` field to deploy and migration request. 
- [ ] Validate if the billingMode provided is supported by the service template. 
- [ ] If not supported then return `BillingModeNotSupported` exception.

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1590
related to eclipse-xpanse/xpanse-ui#774

### CODE REVIEW

**Comment by Alice1319** [comment]
LGTM

**Review by baixinsui — APPROVED** [review]
LGTM

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java [hunk 1]
```diff
@@ -120,6 +121,11 @@ private void validateEula(MigrateRequest migrateRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(migrateRequest.getBillingMode())) {
+            log.error("The service template does not support this billing mode.");
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java:126]: It is better to output the template id and billingMode in the error message and then throw the error message.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 2]
```diff
@@ -107,6 +108,11 @@ public DeployTask createNewDeployTask(DeployRequest deployRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(deployRequest.getBillingMode())) {
+            log.error("The service template does not support this billing mode.");
+            throw new BillingModeNotSupported("BillingMode not supported.");
+        }
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:115]: The error message should be updated in the same way above.

**Comment by baixinsui** [comment]
@WangLiNaruto You can fix the test errors by setting billingMode in the method `getDeployRequest(ServiceTemplateVo templete)`  in the  calss`ApisTestCommon`as below:

**Comment by Alice1319** [comment]
For test errors, we can check the test .yml file, then refine the relevant test data and try again

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java (hunk 1 above)
Comment (Alice1319) [reply modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java:126]: Good advice.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java (hunk 2 above)
Comment (Alice1319) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:115]: Good advice.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 3]
```diff
@@ -107,6 +108,11 @@ public DeployTask createNewDeployTask(DeployRequest deployRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(deployRequest.getBillingMode())) {
+            log.error("The service template does not support this billing mode.");
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:113]:
```suggestion
            String errorMsg = String.format(
                    "The service template with id %s does not support this billing mode %s.",
                    existingServiceTemplate.getId(), deployRequest.getBillingMode());
             log.error(errorMsg);
```

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java [hunk 4]
```diff
@@ -120,6 +121,15 @@ private void validateEula(MigrateRequest migrateRequest) {
...
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(migrateRequest.getBillingMode())) {
+            log.error("The service template with id {} does not support this billing mode {}.",
+                    existingServiceTemplate.getId(), migrateRequest.getBillingMode());
+            String errorMsg = String.format(
+                    "The service template with id %s does not support this billing mode %s.",
+                    existingServiceTemplate.getId(), migrateRequest.getBillingMode());
+            throw new BillingModeNotSupported(errorMsg);
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java:131]:
```suggestion
            log.error(errorMsg)
            throw new BillingModeNotSupported(errorMsg);
```

File: modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java [hunk 5]
```diff
@@ -120,6 +121,15 @@ private void validateEula(MigrateRequest migrateRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(migrateRequest.getBillingMode())) {
+            log.error("The service template with id {} does not support this billing mode {}.",
```
Comment (baixinsui) [inline modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceMigrationApi.java:126]:
```suggestion
```

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 6]
```diff
@@ -107,6 +108,11 @@ public DeployTask createNewDeployTask(DeployRequest deployRequest) {
...
+            throw new BillingModeNotSupported("BillingMode not supported.");
```
Comment (baixinsui) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:114]:
```suggestion
            throw new BillingModeNotSupported(errorMsg);
```

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 7]
```diff
@@ -107,6 +108,14 @@ public DeployTask createNewDeployTask(DeployRequest deployRequest) {
             log.error("Service not accepted Eula.");
             throw new EulaNotAccepted("Service not accepted Eula.");
         }
+        if (!existingServiceTemplate.getOcl().getBilling().getBillingModes()
+                .contains(deployRequest.getBillingMode())) {
+            String errorMsg = String.format(
+                    "The service template with id %s does not support this billing mode %s.",
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:114]: we can remove the word 'this' in this error message.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**